lisa/sut_orchestrator/azure/notifiers.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import json
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Type
+
+from lisa import messages, notifier, schema
+from lisa.environment import EnvironmentMessage, EnvironmentStatus
+from lisa.util import LisaException, constants
+
+from .. import AZURE
+from .common import AzureNodeSchema
+
+
+@dataclass
+class VmsizeBasicInfo:
+    name: str = ""
+    vm_family: str = ""
+    core_count: int = 0
+
+
+@dataclass
+class EnvironmentVmsizeInfo:
+    name: str
+    status: EnvironmentStatus = EnvironmentStatus.New
+    vm_sizes: List[VmsizeBasicInfo] = field(default_factory=list)
+
+
+class AzureNotifier(notifier.Notifier):
+    """
+    This notifier is used to collect the core usage information, which can then guide
+    the quota request in the subscription.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "azure_notifier"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return schema.Notifier
+
+    def _received_message(self, message: messages.MessageBase) -> None:
+        if isinstance(message, EnvironmentMessage):
+            self._process_environment_message(message)
+        else:
+            raise LisaException(f"unsupported message received, {type(message)}")
+
+    def _subscribed_message_type(self) -> List[Type[messages.MessageBase]]:
+        return [EnvironmentMessage]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        env_path = constants.RUN_LOCAL_LOG_PATH / "environments"
+        env_path.mkdir(exist_ok=True, parents=True)
+        self._core_stats_file_path = env_path / "core_stats.json"
+        self._collection_file_path = env_path / "core_stats_collection.log"
+        self._environments: Dict[str, EnvironmentVmsizeInfo] = {}
```

Comment:
The env looks not useful, just save the results. So it should be removed.

Reply:
I don't think so. For skipped runs that are caused by no available environment, it has a deleted message but no deploy message. The deleted message can be ignored if there is no env cache data

Reply:
I see. It doesn't need to save the information, just use a `set` called `_created_envs`